### PR TITLE
Add Solarus

### PR DESCRIPTION
**Pull request by Mailaender**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://gitlab.com/solarus-games/solarus-flatpak/-/issues/3
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. https://gitlab.com/solarus-games/solarus/-/merge_requests/1417

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by Mailaender** [comment]
bot, build org.solarus_games.solarus

**Comment by Mailaender** [comment]
bot, build org.solarus_games.solarus

**Comment by Mailaender** [comment]
bot, build org.solarus_games.solarus

**Comment by Mailaender** [comment]
bot, build org.solarus_games.solarus

**Comment by hfiguiere** [comment]
The screenshot referenced in the appstream file is 404...

**Comment by Mailaender** [comment]
bot, build org.solarus_games.solarus

**Comment by Mailaender** [comment]
bot, build org.solarus_games.solarus

**Comment by Mailaender** [comment]
bot, build org.solarus_games.solarus

**Comment by barthalion** [comment]
You need to cross-check the desktop and the appdata file for ID usage, `<launchable>` tag and whether icon exists at all.

**Comment by Mailaender** [comment]
bot, build org.solarus_games.solarus

**Comment by Mailaender** [comment]
bot, build org.solarus_games.solarus.Launcher

**Comment by Mailaender** [comment]
bot, build org.solarus_games.solarus.Launcher

File: org.solarus_games.solarus.Launcher.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+finish-args:
+  - --share=ipc
+  - --socket=wayland
+  - --socket=fallback-x11
+  - --socket=pulseaudio
+  - --device=dri
+
+modules:
+  - glm.json
+  - org.luajit.Luajit.json
```
Comment (barthalion) [inline org.solarus_games.solarus.Launcher.yaml:15]: Please merge these back to the main manifest.

Comment (Mailaender) [reply org.solarus_games.solarus.Launcher.yaml:15]: Can you elaborate why and how?

Comment (hfiguiere) [reply org.solarus_games.solarus.Launcher.yaml:15]: 1. open `org.luajit.Luajit.json` in text editor. Select all, copy. 
2. in the main manifest, Paste, Reindent as needed by YAML.

Comment (Mailaender) [reply org.solarus_games.solarus.Launcher.yaml:15]: But the `glm.json` is a shared module and LuaJIT probably should be one as well.

Comment (barthalion) [reply org.solarus_games.solarus.Launcher.yaml:15]: It's not a shared module if it's not in the shared-modules repository.

Comment (Mailaender) [reply org.solarus_games.solarus.Launcher.yaml]: Looking at https://luajit.org/download.html I see there are conflicting versions and game makers attempting to use a development version. Looking at https://gitlab.com/solarus-games/solarus/-/blob/dev/compilation.md I was also packaging the wrong one. Probably not a good candiate for a shared version.

**Comment by Mailaender** [comment]
bot, build org.solarus_games.solarus.Launcher

**Comment by Mailaender** [comment]
bot, build org.solarus_games.solarus

**Comment by Mailaender** [comment]
bot, cancel

**Comment by Mailaender** [comment]
bot, build org.solarus_games.solarus.Launcher

**Pull request closed without merging**